libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -235,7 +235,12 @@ proc handleSubscribe*(g: GossipSub,
 
     # unsubscribe remote peer from the topic
     g.gossipsub.removePeer(topic, peer)
-    g.mesh.removePeer(topic, peer)
+
+    if g.mesh.hasPeer(topic, peer):
+      #against spec
+      g.mesh.removePeer(topic, peer)
+      g.pruned(peer, topic)
```

Comment:
this also makes me wonder: should we move `mesh.removePeer` to `pruned`?

Reply:
In some cases, pruned is called without removePeer (eg, when we unsub, we simply remove the topic from mesh)
Though this pruned is definitively error prone, has shown here

We could do it the other way around, add a custom event in the PeerTable which calls pruned for GS
But pubsub & gossipsub needs a refacto anyway